src/actions/transfer.ts

```diff
@@ -82,12 +126,25 @@ const buildTransferDetails = async (
   });
 
   const parsedXml = parseKeyValueXml(xmlResponse);
+  
 
   if (!parsedXml) {
-    throw new Error('Failed to parse XML response from LLM for transfer details.');
+    throw new Error(
+      "Failed to parse XML response from LLM for transfer details.",
+    );
   }
 
-  const transferDetails = parsedXml as unknown as TransferParams;
+  const transferDetails = {
+    ...parsedXml,
+    fromChain: parsedXml.fromChain,
+    recipientAddress: getAddress(parsedXml.recipientAddress),
+    tokenDecimals: Number(parsedXml.tokenDecimals),
+    toAddress: getAddress(parsedXml.toAddress),
+    token: parsedXml.token
+  } as TransferParams;
+
+  transferDetails.amount = BigInt(parseUnits(parsedXml.amount, transferDetails.tokenDecimals));
+  transferDetails.data = await getTransferData(transferDetails.amount, transferDetails.recipientAddress);
```

Comment:
```suggestion
  transferDetails.data = await getTransferCalldata(transferDetails.amount, transferDetails.recipientAddress);
```

Rename the function name